client/web/src/enterprise/insights/components/insights-view-grid/components/backend-insight/components/drill-down-filters-action/DrillDownFiltersPanel.tsx

```diff
@@ -45,26 +43,23 @@ export const DrillDownFiltersAction: React.FunctionComponent<DrillDownFiltersPro
     const isFiltered = hasActiveFilters(initialFiltersValue)
 
     return (
-        <>
-            <Button
+        <Popover open={isOpen} anchor={popoverTargetRef} onOpenChange={event => onVisibilityChange(event.isOpen)}>
+            <PopoverTrigger
+                as={Button}
                 ref={targetButtonReference}
-                className={classNames('btn-icon p-1', styles.filterButton, {
+                type="button"
+                aria-label={isFiltered ? 'Active filters' : 'Filters'}
+                className={classNames('btn btn-icon p-1', styles.filterButton, {
```

Comment:
The `btn` class is redundant with `as={Button}`

```suggestion
                className={classNames('btn-icon p-1', styles.filterButton, {
```